vowpalwabbit/core/include/vw/core/learner.h

```diff
@@ -329,6 +337,13 @@ class learner
     if (_save_load_fd.base) { _save_load_fd.base->save_load(io, read, text); }
   }
 
+  // called to edit the command-line from a reduction.
+  inline void NO_SANITIZE_UNDEFINED output_single_model(VW::workspace& all)
+  {
+    if (_output_single_model_fd.output_single_model_f == nullptr) { return; }
+    _output_single_model_fd.output_single_model_f(all, _output_single_model_fd.data);
```

Comment:
shouldn't you call recursively down the stack in case another reduction needs to do some cleanup? might be an extreme edge case for now - but similar to save_load.

Reply:
Updated to be recursive. I dont think it makes sense at the moment to have more than one of these in the stack, but could be useful in the future